### PR TITLE
Handle Block Construction Signal and Broadcast

### PR DESCRIPTION
**Pull request by tholonious**

### CODE REVIEW

**Comment by tholonious** [comment]
Current dependencies on/for this PR:
* feature/pos-networking-and-syncing
  * **PR #811** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/811?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/811?utm_source=stack-comment).

File: lib/pos_consensus_controller.go [hunk 1]
```diff
@@ -37,21 +41,109 @@ func (cc *ConsensusController) Init() {
...
+		// If the event is not properly formed, we ignore it and log it. This should never happen.
+		return errors.Errorf("HandleFastHostStuffBlockProposal: Received improperly formed vote QC construction event: %v", event)
+	}
+
+	// If the block proposal is properly formed, we try to construct and broadcast the block here.
+
+	// Fetch the parent block
+	parentBlockHash := BlockHashFromConsensusInterface(event.QC.GetBlockHash())
+	parentBlock, parentBlockExists := cc.blockchain.blockIndex[*parentBlockHash]
+	if !parentBlockExists {
```
Comment (lazynina) [inline lib/pos_consensus_controller.go:69]: Eventually we should add a `IsValidated()` check probably?

File: lib/pos_consensus_controller.go [hunk 2]
```diff
@@ -435,3 +526,7 @@ func (epochEntry *EpochEntry) ContainsBlockHeight(blockHeight uint64) bool {
 func (bc *Blockchain) getSafeBlocks() ([]*MsgDeSoHeader, error) {
 	return nil, errors.New("getSafeBlocks: replace me with a real implementation later")
 }
+
+func (bav *UtxoView) getNextRandomSeedHash() (*RandomSeedHash, error) {
```
Comment (lazynina) [inline lib/pos_consensus_controller.go:530]: I'll have this for you in the `feature/pos-process-block` branch.

**Review by lazynina — APPROVED** [review]
LGTM

**Pull request merged**